### PR TITLE
Selenopotential to degree and order 50

### PR DESCRIPTION
**Pull request by eggrobin**
Bump `max_geopotential_degree` and extend the numerical tables accordingly.

### CODE REVIEW

File: physics/geopotential_body.hpp [hunk 1]
```diff
@@ -624,6 +624,7 @@ Geopotential<Frame>::GeneralSphericalHarmonicsAcceleration(
     PRINCIPIA_CASE_SPHERICAL_HARMONICS(8);
     PRINCIPIA_CASE_SPHERICAL_HARMONICS(9);
     PRINCIPIA_CASE_SPHERICAL_HARMONICS(10);
+    PRINCIPIA_CASE_SPHERICAL_HARMONICS(50);
```
Comment (pleroy) [inline physics/geopotential_body.hpp]: As discussed, let's keep 11..49.

**Comment by eggrobin** [comment]
With the new geopotential tolerances, the tests take as long as they used to, but the build still takes more than 30 min; we can't merge this as-is.

**Comment by eggrobin** [comment]
For future reference: the detemplatization of `DegreeNOrderM` is at https://github.com/eggrobin/Principia/tree/detemplatization.

File: astronomy/sol_gravity_model.proto.txt [hunk 2]
```diff
@@ -276,8 +280,1429 @@ gravity_model {
     axis_declination        : "66.5392 deg"
     reference_angle         : "38.3213 deg"
     angular_frequency       : "13.17635815 deg / d"
-    j2                      : 2.0321568464952570e-04
     reference_radius        : "1738.0 km"
+    geopotential {
+      row {
+        degree: 2
+        column {order:  0, cos: -9.0884339347424299e-05, sin:  0.0000000000000000e+00},
```
Comment (pleroy) [inline astronomy/sol_gravity_model.proto.txt]: Could we use `0` for the zeroes, like we did for the Earth?

File: astronomy/sol_gravity_model.proto.txt [hunk 3]
```diff
@@ -96,87 +96,87 @@ gravity_model {
...
         column {order: 5, cos: -4.9285509081565e-08, sin: -5.0617431292160e-08},
         column {order: 6, cos: -3.7586959404344e-08, sin: -7.9771481050624e-08},
-        column {order: 7, cos: 8.2571270822778e-09, sin: -3.0490009250716e-09},
-        column {order: 8, cos: 4.0589757557591e-08, sin: -9.1710645675513e-08},
-        column {order: 9, cos: 1.2538547435906e-07, sin: -3.7942358311393e-08},
-        column {order: 10, cos: 1.0042327725658e-07, sin: -2.3863826960514e-08}},
+        column {order: 7, cos:  8.2571270822778e-09, sin: -3.0490009250716e-09},
+        column {order: 8, cos:  4.0589757557591e-08, sin: -9.1710645675513e-08},
+        column {order: 9, cos:  1.2538547435906e-07, sin: -3.7942358311393e-08},
+        column {order: 10, cos:  1.0042327725658e-07, sin: -2.3863826960514e-08}},
```
Comment (pleroy) [inline astronomy/sol_gravity_model.proto.txt]: This one is not aligned.  There should be an extra space after `order:` for most of the entries, like you did for the Moon.

File: numerics/max_abs_normalized_associated_legendre_functions_test.cc [hunk 4]
```diff
@@ -53,5 +54,16 @@ TEST_F(MaxAbsNormalizedAssociatedLegendreFunctionTest, HighestOrder) {
   }
 }
 
+TEST_F(MaxAbsNormalizedAssociatedLegendreFunctionTest, Monotonicity) {
+  for (int m = 0; m < MaxAbsNormalizedAssociatedLegendreFunction.rows; ++m) {
+    for (int n = m; n < MaxAbsNormalizedAssociatedLegendreFunction.rows - 1;
```
Comment (pleroy) [inline numerics/max_abs_normalized_associated_legendre_functions_test.cc]: Line break after `;`.

**Comment by eggrobin** [comment]
Benchmarking: the loss of the `FORCE_INLINE` doesn't change much, it just gets slower regardless, see https://gist.github.com/eggrobin/3453c16bf67f9a99f3e24c57722eeeda/revisions.

File: astronomy/sol_gravity_model.proto.txt (hunk 2 above)
Comment (eggrobin) [reply astronomy/sol_gravity_model.proto.txt]: Done.

File: astronomy/sol_gravity_model.proto.txt (hunk 3 above)
Comment (eggrobin) [reply astronomy/sol_gravity_model.proto.txt]: Done.

**Pull request merged**